biogeochem/EDPhysiologyMod.F90

```diff
@@ -1084,154 +1111,333 @@ subroutine phenology( currentSite, bc_in )
+                           real(numWaterMem,r8)
+       mean_10day_smp    = sum(currentSite%smp_memory   (1:numWaterMem,ipft)) / &
+                           real(numWaterMem,r8)
+
+       ! Compare the moisture with the threshold.
+       if ( ED_val_phen_drought_threshold >= 0. ) then
+          ! Liquid volume in reference layer (m3/m3)
+          smoist_below_threshold = mean_10day_liqvol < ED_val_phen_drought_threshold
+       else
+          ! Soil matric potential in reference layer (mm)
+          smoist_below_threshold = mean_10day_smp    < ED_val_phen_drought_threshold
+       end if
 
-    if(currentSite%dstatus == phen_dstat_timeoff ) then
-       if (dayssincedleafoff > min_daysoff_dforcedflush) then
-          currentSite%dstatus     = phen_dstat_timeon    ! force budburst!
-          currentSite%dleafondate = model_day_int        ! record leaf on date
-          dayssincedleafon        = 0
+       ! Calculate days since last flushing and shedding event, but make a provision
+       ! for the first year of simulation, we have to assume leaf drop / leaf flush
+       ! dates to start, so if that is in the future, set it to last year
+       if (model_day_int < currentSite%dleafoffdate(ipft)) then
+          currentSite%dndaysleafoff(ipft) = model_day_int - (currentSite%dleafoffdate(ipft)-365)
+       else
+          currentSite%dndaysleafoff(ipft) = model_day_int - currentSite%dleafoffdate(ipft)
+       end if
+       if (model_day_int < currentSite%dleafondate(ipft)) then
+          currentSite%dndaysleafon(ipft) = model_day_int - (currentSite%dleafondate(ipft)-365)
+       else
+          currentSite%dndaysleafon(ipft) = model_day_int - currentSite%dleafondate(ipft)
        end if
-    end if
 
-    ! LEAF OFF: DROUGHT DECIDUOUS LIFESPAN - if the leaf gets to
-    ! the end of its useful life. A*, E*
-    ! i.e. Are the leaves rouhgly at the end of their lives?
 
-    if ( (currentSite%dstatus == phen_dstat_moiston .or. &
-         currentSite%dstatus == phen_dstat_timeon ) .and. &
-         (dayssincedleafon > canopy_leaf_lifespan) )then
-       currentSite%dstatus      = phen_dstat_timeoff    !alter status of site to 'leaves off'
-       currentSite%dleafoffdate = model_day_int         !record leaf on date
-    endif
-
-    ! LEAF OFF: DROUGHT DECIDUOUS DRYNESS - if the soil gets too dry,
-    ! and the leaves have already been on a while...
+       ! Elongation factor from the previous step.
+       elongf_prev = currentSite%elong_factor(ipft)
+
+
+       ! PFT leaf lifespan in days. This is the shortest between the leaf longevity 
+       ! (defined as a PFT parameter) and the maximum canopy leaf life span allowed 
+       ! for drought deciduous (local parameter).
+       ! MLO: Need to double check that this will not cause unintended problems
+       pft_leaf_lifespan = min( canopy_leaf_lifespan, &
+                                nint(365*maxval(prt_params%leaf_long(ipft,:))) )
```

Comment:
Flagging this part for revision...

Variable `canopy_leaf_lifespan` (365 days) is defined locally and used to represent the leaf longevity and decide when to force PFTs to shed leaves. However, the leaf lifespan (as defined in the parameter file) can be shorter than one year, so I assumed the time for forced shedding should be the shorter between the parameter longevity and `canopy_leaf_lifespan`.

Reply:
I think this makes sense.  I would only suggest that we explain the dual meaning in the parameter file and the parameter definition in-code.
https://github.com/NGEET/fates/blob/b570b36f0badecbfd7533af7747cafb3194eb94a/parameter_files/fates_params_default.cdl#L550

Reply:
Also, FYI: the leaf lifespan also impacts leaf aging.  My sense is that many simulations out there do not use variable leaf age impacts on vcmax, but it is an option. I'm not saying what you are proposing is in conflict with it, just wanted to point it out.
https://github.com/NGEET/fates/blob/main/parteh/PRTGenericMod.F90#L1312

Reply:
> I think this makes sense. I would only suggest that we explain the dual meaning in the parameter file and the parameter definition in-code.
> 
> https://github.com/NGEET/fates/blob/b570b36f0badecbfd7533af7747cafb3194eb94a/parameter_files/fates_params_default.cdl#L550

Sure, I will add a brief explanation in both places. By the way, I find the parameter name `fates_turnover_leaf` slightly confusing, because it reminds me of leaf turnover rate (yr<sup>-1</sup>)...

Reply:
> Also, FYI: the leaf lifespan also impacts leaf aging. My sense is that many simulations out there do not use variable leaf age impacts on vcmax, but it is an option. I'm not saying what you are proposing is in conflict with it, just wanted to point it out.
> https://github.com/NGEET/fates/blob/main/parteh/PRTGenericMod.F90#L1312

Thanks for the heads up, but my impression is that this change would not interfere with the leaf age impacts on Vcmax.  The code already forced leaf shedding once the cohort's leaf age reached 1 year. The change here was simply to shorten the leaf "expiration date" to be the maximum leaf longevity across all leaf age classes when this value was < 1 year.

Reply:
shouldn't that be sum() (or something similar) instead of maxval then?

Reply:
Indeed, thanks! I will submit an update shortly.